kafka-rest/src/test/java/io/confluent/kafkarest/integration/AvroProducerTest.java

```diff
@@ -182,20 +185,15 @@ protected <K, V> void testProduceToPartition(
     Response response =
         request("/topics/" + topicName + "/partitions/0", queryParams)
             .post(Entity.entity(payload, Versions.KAFKA_V2_JSON_AVRO));
-    assertOKResponse(response, Versions.KAFKA_V2_JSON);
-    final ProduceResponse poffsetResponse =
-        TestUtils.tryReadEntityOrLog(response, ProduceResponse.class);
-    assertEquals(offsetResponse, poffsetResponse.getOffsets());
-    TestUtils.assertTopicContains(
-        plaintextBrokerList,
-        topicName,
-        payload.toProduceRequest().getRecords(),
-        0,
-        KafkaAvroDeserializer.class.getName(),
-        KafkaAvroDeserializer.class.getName(),
-        deserializerProps,
-        false);
-    assertEquals((Integer) 1, poffsetResponse.getValueSchemaId());
+    assertErrorResponse(
+        ConstraintViolationExceptionMapper.UNPROCESSABLE_ENTITY,
+        response,
+        42201,
+        null,
+        Versions.KAFKA_V2_JSON);
+    ErrorResponse actual = response.readEntity(ErrorResponse.class);
+    assertEquals(42201, actual.getErrorCode());
+    assertEquals("Unrecognized field: error_code", actual.getMessage());
```

Comment:
you mentioned this test was failing, but if the error code is ok, i think we can ignore the message check?

I'm confused about the message too "Request includes keys but does not include key schema" since this test does not provide keys. But perhaps that is expected since we took out the "data is null" check right before the `!schema.IsPresent` check in `SchemaRecordSerializerImpl.serialize`. Also, we didn't register an key-schema w/ the schema-registry client, so the serializer doesn't know which schema to check against

Reply:
Agreed on the null schema/null payload thing : that should be ok.

Hritwik pulled together a doc showing the valid combinations, now linked in the PR description